src/environments/array.js

```diff
@@ -46,6 +60,18 @@ function parseArray(
     let row = [];
     const body = [row];
     const rowGaps = [];
+    const numHLinesBeforeRow = [];
+
+    // Test for \hline at the top of the array.
+    if (nextTokenIsHLine(parser)) {
+        numHLinesBeforeRow.push(1);
+        if (nextTokenIsHLine(parser)) {
+            numHLinesBeforeRow[0] = 2;
+        }
+    } else {
```

Comment:
I think the following is more readable (same for below), but it's matter of personal preference:
```js
    if (nextTokenIsHLine(parser)) {
        if (nextTokenIsHLine(parser)) {
            numHLinesBeforeRow.push(2);
        } else {
            numHLinesBeforeRow.push(1);
        }
    } else {
```

Reply:
Is there something magic about 2? Why not a `while` loop and counter? Does LaTeX not support more than 2 `\hline`s?

Reply:
Until just now, I had never tried more than two `\hline`s. `while` loop coming right up.